test/Microsoft.AspNetCore.Razor.Design.Test/IntegrationTests/MSBuildIntegrationTestBase.cs

```diff
@@ -29,28 +19,19 @@ protected MSBuildIntegrationTestBase()
         // Used by the test framework to set the project that we're working with
         internal static ProjectDirectory Project
         {
-#if NET461
-            get
-            {
-                var handle = (ObjectHandle)CallContext.LogicalGetData("MSBuildIntegrationTestBase_Project");
-                return (ProjectDirectory)handle.Unwrap();
-            }
-            set
-            {
-                CallContext.LogicalSetData("MSBuildIntegrationTestBase_Project", new ObjectHandle(value));
-            }
-#elif NETCOREAPP2_0 || NETCOREAPP2_1
             get { return _project.Value; }
             set { _project.Value = value; }
-#else
-#error TFM not supported
-#endif
         }
 
-        internal Task<MSBuildResult> DotnetMSBuild(string target, string args = null, bool debug = false)
+        internal Task<MSBuildResult> DotnetMSBuild(string target, string args = null, bool restore = true, bool debug = false)
         {
             var timeout = debug ? (TimeSpan?)Timeout.InfiniteTimeSpan : null;
-            return MSBuildProcessManager.RunProcessAsync(Project, $"/t:{target} {args}", timeout);
+            var msbuildArgs = $"/t:{target} {args}";
+            if (restore)
+            {
+                msbuildArgs = "/restore " + msbuildArgs;
```

Comment:
I think they recommend doing this now. I can't find the email thread, but there were some scenarios where they advise against calling the target and instead using this switch. We could make the parameter `false` by default.

Reply:
no, I think this is good cleanup